homeassistant/components/smartthings/climate.py

```diff
@@ -454,11 +488,65 @@ def hvac_modes(self) -> list[HVACMode]:
         return self._hvac_modes
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> float:
         """Return the temperature we try to reach."""
         return self._device.status.cooling_setpoint
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return the unit of measurement."""
-        return UNIT_MAP.get(self._device.status.attributes[Attribute.temperature].unit)
+        return (
+            UNIT_MAP.get(self._device.status.attributes[Attribute.temperature].unit)
+            or ""
+        )
```

Comment:
```suggestion
        return (
            UNIT_MAP[self._device.status.attributes[Attribute.temperature].unit]
        )
```
Even better to move this to init and set it once.

Reply:
why? If I do this and the smartthings API returns unexpected unit, the code would fail. Isn't it better the way how it is now?

Reply:
You can leave it if it can change between c and f but if it can also return something unexpected I would be slightly worried about it.

Reply:
ok, you've got a point. Changed